pkg/apply/desiredset_process.go

```diff
@@ -494,3 +498,45 @@ func multiNamespaceList(ctx context.Context, namespaces []string, baseClient dyn
 
 	return wg.Wait()
 }
+
+// canIndexByHash detects if provided selector can be replaced by using the hash index, if configured, in which case returns the hash value
+func canIndexByHash(indexer cache.Indexer, selector labels.Selector) (string, bool) {
```

Comment:
nit:
You can ignore this because it is opinionated and not a big deal, but this function caused some confusion for me when I was first reading the code.
prefixes such as `can`, `is`, and `has` usually imply that we are returning a bool, while this function is more of a retrieval function that lets you know if the value was found or not.
```suggestion
func getIndexByHash(indexer cache.Indexer, selector labels.Selector) (string, bool) {
```